src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -125,15 +149,25 @@ func (server *Server) ListenAndServe() error {
 		return fmt.Errorf("Unable to listen at %s: %s", server.Addr, err)
 	}
 
-	go server.register()
+	go server.register(updateConfig)
 
 	return fs.Serve(l)
 }
 
-func (server *Server) register() {
+func (server *Server) register(updateConfig func(func(*ServerConfig) error)) {
+	supportedFronts := make([]string, 0, len(frontingProviders))
+	for name := range frontingProviders {
+		supportedFronts = append(supportedFronts, name)
+	}
 	for {
 		server.cfgMutex.RLock()
 		baseUrl := server.cfg.RegisterAt
+		var port string
+		if server.cfg.Unencrypted {
```

Comment:
I just realised we should really get the port from the `-addr` value, but then we'd need to make `-unencrypted` take another port (and perhaps another host).  I'll think about this in the context of implementing dual listening.  In the meanwhile, this will work for our own fallbacks, which at the moment are the only ones peerscanner is registering.